install/tests/Makefile

```diff
@@ -37,286 +21,317 @@ help: Makefile
-cert-manager:
-	$(MAKE) select-workspace && \
-	terraform apply -target=module.certmanager -var kubeconfig=${KUBECONFIG} --auto-approve
-	@echo "Done installing cert-manager"
-
-.PHONY:
-## managed-dns: Installs external-dns, and setup up CloudDNS access
-managed-dns: check-env-sub-domain select-workspace
-	terraform apply -target=module.clouddns-externaldns -var kubeconfig=${KUBECONFIG} --auto-approve
-	@echo "Done created GCP managed DNS"
-
-.PHONY:
-## external-dns: Installs external-dns
-external-dns: check-env-cloud select-workspace
-	terraform apply -target=module.$(cloud)-externaldns -var kubeconfig=${KUBECONFIG} --auto-approve
-	@echo "Done creating externaldns for $(cloud)"
 
+## TODO: (nvn) cleanup all the `get-kubeconfig` targets
 .PHONY:
 ## get-kubeconfig: Returns KUBECONFIG of a just created cluster
 get-kubeconfig:
-	echo "Getting kubeconfig for $$TF_VAR_TEST_ID terraform state" && \
+	@echo "Getting kubeconfig for $$TF_VAR_TEST_ID terraform state" && \
     export provider=$$(echo "$$TF_VAR_TEST_ID" | sed 's/\(.*\)-/\1 /' | xargs | awk '{print $$2}') && \
+	export provider=k3s && \
 	$(MAKE) $$provider-kubeconfig && echo "kubeconfig written to ${KUBECONFIG}"
 
+sync-kubeconfig:
+	[[ -z $$self_hosted_jobs ]] || gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS} --project=sh-automated-tests
+	gcloud config set project sh-automated-tests
+	gsutil cp gs://nightly-tests/tf-state/${TF_VAR_TEST_ID}-kubeconfig ${KUBECONFIG} || echo "No kubeconfig"
+
+## k3s-kubeconfig: Get the kubeconfig configuration for GCP K3s
+k3s-kubeconfig: sync-kubeconfig
+
+## gcp-kubeconfig: Get the kubeconfig configuration for GCP GKE
+gcp-kubeconfig:
+	@[[ -n $$self_hosted_jobs ]] || $(MAKE) download-creds
+	@[[ -n $$self_hosted_jobs ]] || cat $(TF_VAR_TEST_ID)-output.json | yq r - 'gke_user_key.value' | tr -d '\n' | jq -Rsa . > ${TF_VAR_TEST_ID}-key.json
+	@[[ -f ${TF_VAR_TEST_ID}-key.json ]] || cat ${GOOGLE_APPLICATION_CREDENTIALS} > ${TF_VAR_TEST_ID}-key.json
+	@gcloud auth activate-service-account --key-file=${TF_VAR_TEST_ID}-key.json --project=sh-automated-tests || { echo "Count not authenicate the service account"; exit 1; }
+	@export KUBECONFIG=${KUBECONFIG} && \
+	gcloud container clusters get-credentials gp-${TF_VAR_TEST_ID} --zone europe-west1-d --project sh-automated-tests || echo "No cluster present"
+	@rm -f ${TF_VAR_TEST_ID}-key.json
+	@echo "Kubeconfig written to ${KUBECONFIG}"
+
+## azure-kubeconfig: Get the kubeconfig configuration for Azure AKS
+azure-kubeconfig:
+	@[[ -z "$$self_hosted_jobs" ]] || az login --service-principal -u $$ARM_CLIENT_ID -p $$ARM_CLIENT_SECRET  --tenant $$ARM_TENANT_ID
+	@export KUBECONFIG=${KUBECONFIG} && \
+	az aks get-credentials --name gp-$$TF_VAR_TEST_ID-cluster --resource-group gp-$$TF_VAR_TEST_ID --file ${KUBECONFIG} || echo "No cluster present"
+
+get-aws-secret:
+	@gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS} --project=sh-automated-tests
```

Comment:
**suggestion(non-blocking, out-of-scope):** We reactivate service account credentials in a number of makefile targets, and also diffuses the responsibility of providing credentials across a bunch of different make targets. What do you think of having a make target like `activate-credentials` that runs at the beginning of all our test configurations? Moving this logic to a single target and using `installer-tests.ts` to execute the makefile target ensures we have a consistent environment for the rest of the test execution, decouples environment setup from the individual tests, and gives us a clear point where a user manually running make targets can opt-in (or opt-out) of this activation behavior.

Non-blocker, no change required; out of scope; this is a non-trivial change that we can address in another pull request.